### PR TITLE
docs update for 3.17 release, ChangeLog and ReleaseNotes

### PR DESCRIPTION
**Pull request by drelaptop**

### CODE REVIEW

**Comment by drelaptop** [comment]
ChangeLog finished depend on the closed issues and PRs marked 3.17 milestone, could you help me review it @minggo @slackmoehrle , are the __HIGHTLIGHT__ items proper, or other ideas?

File: CHANGELOG [hunk 1]
```diff
@@ -1,3 +1,105 @@
+cocos2d-x-3.17 May 2018
```
Comment (minggo) [inline CHANGELOG]: Should change to exactly day before releasing.

File: CHANGELOG [hunk 2]
```diff
@@ -1,3 +1,105 @@
...
+
+[HIGHLIGHT]     Android: support Android Studio 3.0, NDK 16
```
Comment (minggo) [inline CHANGELOG]: It is Android Studio 3.0 or 3.1?

**Comment by minggo** [comment]
Yep, `HIGHLIGHTS` is good here.

File: CHANGELOG (hunk 2 above)
Comment (drelaptop) [reply CHANGELOG]: Our test team tested the Android project using 3.0, and the default Gradle config is suitable for 3.0. So I think we should tell developers we support 3.0, even though some developers said that the project goes well in 3.01, 3.1 only need to update the Gradle config.

File: CHANGELOG (hunk 1 above)
Comment (drelaptop) [reply CHANGELOG]: yep, I will do it

**Comment by drelaptop** [comment]
Below is my draft of the English highlights for 3.17 release, could you review this @slackmoehrle ,the highlights will be used in cocos2d-x.org website.
```
support iPhone X
support Android Studio 3.0
CMake support all platforms
upgrade Spine runtime to v3.6.39 
upgrade GLFW to 3.2.1
bugs fix, some improvements
```
> and this is the Chinese highlights, we prepare to use:
```
支持 iPhone X
支持 Android Studio 3.0
扩展 CMake 构建，支持所有平台
升级 Spine runtime 到 v3.6.39 
升级 GLFW 到 3.2.1
bugs 修复，提高稳定性

```

File: CHANGELOG (hunk 2 above)
Comment (minggo) [reply CHANGELOG]: Sounds good, can mention it in release note.

File: docs/RELEASE_NOTES_CN.md [hunk 3]
```diff
@@ -2,102 +2,81 @@
...
+### 支持 Android Studio 3.0
 
-LayerRadialGradien和LayerColor类似，它会在制定的圆里绘制颜色，效果如下：
+Android Studio 是一个成熟的，同时是 Google 官方支持的 Android 开发 IDE，引擎跟进支持。在支持 3.0 的过程中，同时将支持的 NDK 版本从 r14 升级到 r16。Gradle 工程配置方面，简化 Gradle 变量 PROP_*；将依赖声明中弃用的 compile 改为新的 implementation；增加用以削减 Release 包体积的 Proguard 配置。
 
-![][6]
+### CMake 支持全平台
 
-### Lua工程在Xcode 8.0+模拟器崩溃问题
+扩展 CMake 构建支持至全平台，包括 iOS、Android、macOS、Linux、Windows（VC++ compiler）。支持将引擎部分预编译，并在新的构建过程中重用预编译的引擎库。通过使用预编译库，可以极大的缩短工程的构建时间。
```
Comment (minggo) [inline docs/RELEASE_NOTES_CN.md]: It is better to add a link of CMake usage docs here.

File: docs/RELEASE_NOTES_CN.md [hunk 4]
```diff
@@ -2,102 +2,81 @@
...
-插件的详细使用方法可以参考插件仓库的[the READE][1]，欢迎使用并给与反馈。
+## 详细介绍
 
-![][4]
-__Scene effect in cocos creator__
+### 支持 iPhone X
 
-![][5]
-__Scene effect in cocos2d-x__
+自 iPhone X 流行，使用引擎的 iOS 开发者存在升级支持 iPhone X 的需求，该版本为开发者解决了这个问题。支持 iPhone X 全屏，启动屏幕改用 Storyboard，增加了一个获取 SafeArea 的接口，支持自动隐藏 Home 指示器。
```
Comment (minggo) [inline docs/RELEASE_NOTES_CN.md]: It is better to add some pictures to show the feature here. And more detail information here is needed, for example, add a link of apple docs about what's `SafeArea`, how to get it? Add some code snippet here to describe it.

File: docs/RELEASE_NOTES_CN.md [hunk 5]
```diff
@@ -2,102 +2,81 @@
...
+### 升级第三方库
 
-### 修复iOS 11编译错误
+Spine 骨骼动画在使用 Cocos2d-x 开发的游戏中被广泛使用，使用新版的 Spine 导出的素材在老版本的 runtime 上运行存在一些兼容性问题，升级 Spine Runtime 至 3.6.39，解决这些问题。
 
-v3.16版本之前，`CCFileUtils`使用`system()`函数来删除目录。iOS 11删除了`system`，因此导致了编译错误。改版本修复了这个问题。对应的PR链接在[这里](https://github.com/cocos2d/cocos2d-x/pull/17921)，如果不想更新引擎的话，可以参考该PR的改动合并以修复这个问题。
+原 GLFW 版本是 3.2.0，在使用 GLFW 支持游戏手柄时存在问题，升级 GLFW 至 3.2.1 后解决，同时提供 GLFW 预编译库。
 
-### Cocos Studio
+原 Box2D 版本 2.3.1，2014 年发布的版本，截至 2018 年初新的版本一直未发布，但 GitHub 已经有一些 bugs 修复，根据社区的建议，升级 Box2D 至最新提交 f655c603ba9d83，同时提供 Box2D 的预编译库。
```
Comment (minggo) [inline docs/RELEASE_NOTES_CN.md]: It is better to add a link of doc about how to build prebuilt these libs.

File: docs/RELEASE_NOTES_CN.md [hunk 6]
```diff
@@ -2,102 +2,81 @@
...
 
-v3.15更新了flatbuffer，破坏了Cocos Studio reader的兼容性。该版本回退了flatbuffer和Cocos Studio的reader。
+### 移除过时的内容
 
-如果不想更新引擎的话，可以手动从改版本或者v3.15之前的版本拷贝flatbuffer和Cocos Studio的reader。
+由于 Google 官方自 Android SDK Tools 25.3.0 就已放弃 ant 支持，另一方面 Android Studio 已经足够完善，于是决定移除旧有的 ant 工程 `proj.android`，重命名原 Android Studio 工程 `proj.android-studio` 为 `proj.android`。另外 armeabi 已被弃用，默认的 Android 工程编译架构改为 armeabi-v7a。
```
Comment (minggo) [inline docs/RELEASE_NOTES_CN.md]: It is better to add the link of the corresponding google doc here.

**Comment by drelaptop** [comment]
@minggo I will improve it following you said, and add English ReleaseNotes today.

**Comment by drelaptop** [comment]
All finished, Chinese ReleaseNotes existed some changes, please review @minggo , English ReleaseNotes is finished too.

**Comment by minggo** [comment]
Good job.

**Pull request merged**